ynr/account_adapter.py

```diff
@@ -12,3 +17,30 @@ def is_open_for_signup(self, request):
         (Comment reproduced from the overridden method.)
         """
         return False
+
+
+class LoggingSocialAccountAdapter(DefaultSocialAccountAdapter):
+    """
+    Exactly the same as the DefaultSocialAccountAdapter, but logs authentication
+    errors to the default log.
+
+    """
+
+    def authentication_error(
+        self,
+        request,
+        provider_id,
+        error=None,
+        exception=None,
+        extra_context=None,
+    ):
+        """
+        Log errors to authrnticating. This method in the parent class is
+        left blank and exists for overriding, so we can do what we want here.
+
+        """
+        logger.error(
```

Comment:
Will just calling `logger.error()` automatically send this to email/sentry. Do we also need to declare a corresponding handler in

https://github.com/DemocracyClub/yournextrepresentative/blob/9e3db89d5da73b71d5a5de26c13a83ad57c30160/ynr/settings/base.py#L429-L449

to tell django what to do with it?

(I actually can't remember the answer to this)

Reply:
So, I'm quite confused by Django logging in general. 

I think you're right I need to add something, but I think it needs to be added in the deploy repo rather than in this project.

I'll do that now.

Reply:
Yeah that's a good point. If it helps, here's some code in the EE deploy repo that is doing a similar-ish thing:

https://github.com/DemocracyClub/ee_deploy/blob/c489eca9b770295c0994e9496585c68e016acc94/webapp_settings/production.py#L30-L39